helix-core/src/main/java/org/apache/helix/spectator/RoutingTableProvider.java

```diff
@@ -106,61 +115,131 @@ public RoutingTableProvider(HelixManager helixManager, PropertyType sourceDataTy
    */
   public RoutingTableProvider(HelixManager helixManager, PropertyType sourceDataType,
       boolean isPeriodicRefreshEnabled, long periodRefreshInterval) throws HelixException {
-    _routingTableRef = new AtomicReference<>(new RoutingTable());
+    this(helixManager, ImmutableMap.of(sourceDataType, Collections.emptyList()),
+        isPeriodicRefreshEnabled, periodRefreshInterval);
+  }
+
+  /**
+   * Initialize an instance of RoutingTableProvider
+   * @param helixManager
+   * @param sourceDataTypes
+   * @param isPeriodicRefreshEnabled true if periodic refresh is enabled, false otherwise
+   * @param periodRefreshInterval only effective if isPeriodRefreshEnabled is true
+   * @throws HelixException
+   */
+  public RoutingTableProvider(HelixManager helixManager,
+      Map<PropertyType, List<String>> sourceDataTypes, boolean isPeriodicRefreshEnabled,
+      long periodRefreshInterval) throws HelixException {
+
+    // Check and validate the input of the sourceDataTypes parameter
+    for (PropertyType propertyType : sourceDataTypes.keySet()) {
+      if (propertyType.equals(PropertyType.CUSTOMIZEDVIEW)
```

Comment:
nit, Make this a private method?
And maybe we can simplify it little bit. I understand this is for clearly log the error. But we can also put the error message more briefly and log the type/path details in the error log so we can debug.

Reply:
Done.